### ISSUE TITLE
XY Chart: Set Series color when they have the same Y field

### ISSUE THREAD

**Issue by gojanpaolo** [grafana/grafana#95446]
We currently have the following XY Chart Series mapping

`Port` series
![Image](https://github.com/user-attachments/assets/10341175-39a6-4d91-8b94-8973b7ae08f0)

`Stbd` series
![Image](https://github.com/user-attachments/assets/22645bc4-0deb-484b-b472-469cac405e7d)

We would like to have `Port` to be `blue` -- and `Stbd` to be `red`.
We tried configuring the `Color field` and `Field Overrides`. We think the color configurations are coupled to the Y field so it's having trouble when we want to use X field to differentiate the colors/series

**Comment by justinsmalley** [comment]
I am seeing what could be another manifestation of the same issue.

When I set the same field to be used for both color and size, the size stops working, and only the color definitions take effect. I did not see this behaviour in 11.1, but did in 11.3

Workaround was to make a clone of the column so that the color and size were on different columns.

**Comment by gojanpaolo** [comment]
not stale

**Comment by leeoniya** [comment]
> We think the color configurations are coupled to the Y field so it's having trouble when we want to use X field to differentiate the colors/series

this is correct. each series is y-field-bound and uses it for field overrides (otherwise there was no way to make overrides work). i think cloning the field either in the query or via transformation is the correct solution here.

<img width="1229" height="441" alt="Image" src="https://github.com/user-attachments/assets/86fe3834-adb0-4ec6-aba7-dbfebb6279e8" />
